cfme/automate/explorer/domain.py

```diff
@@ -0,0 +1,372 @@
+        if changed:
+            domains_view.flash.assert_message('Priority Order was saved')
+        else:
+            domains_view.flash.assert_message('Edit of Priority Order was cancelled by the user')
+        return changed
+
+
+@navigator.register(DomainCollection)
+class All(CFMENavigateStep):
+    VIEW = DomainListView
+    prerequisite = NavigateToAttribute('appliance.server', 'AutomateExplorer')
+
+    def step(self):
+        self.prerequisite_view.datastore.tree.click_path(*self.obj.tree_path)
+
+
+@navigator.register(DomainCollection)
+class Add(CFMENavigateStep):
+    VIEW = DomainAddView
+    prerequisite = NavigateToSibling('All')
+
+    def step(self):
+        self.prerequisite_view.configuration.item_select('Add a New Domain')
+
+
+@navigator.register(DomainCollection)
+class Priority(CFMENavigateStep):
+    VIEW = DomainPriorityView
+    prerequisite = NavigateToSibling('All')
+
+    def step(self):
+        self.prerequisite_view.configuration.item_select('Edit Priority Order of Domains')
+
+
+class DomainDetailsView(AutomateExplorerView):
+    title = Text('#explorer_title_text')
+    namespaces = Table('#ns_details_grid')
+
+    @property
+    def is_displayed(self):
+        return (
+            self.in_explorer and
+            re.match(
+                r'^Automate Domain "{}(?: \((?:Disabled|Locked|Locked & Disabled)\))?"$'.format(
+                    re.escape(self.context['object'].name)),
+                self.title.text) is not None)
+
+
+class Domain(Navigatable, Fillable):
+    """A class representing one Domain in the UI."""
+    def __init__(
+            self, name, description, enabled=None, locked=None, collection=None, appliance=None):
+        if collection is None:
+            collection = DomainCollection(appliance=appliance)
+        self.collection = collection
+        Navigatable.__init__(self, appliance=collection.appliance)
+        self.name = name
+        self.description = description
+        self.enabled = enabled
+        self.locked = locked
+        self._db_id = None
+
+    def as_fill_value(self):
+        return self.name
+
+    @cached_property
+    def db_id(self):
+        if self._db_id is None:
+            table = self.appliance.db['miq_ae_namespaces']
+            try:
+                self._db_id = self.appliance.db.session.query(table.id).filter(
+                    table.name == self.name,
+                    table.parent_id == None)[0]  # noqa
+            except IndexError:
+                raise ItemNotFound('Domain named {} not found in the database'.format(self.name))
+        return self._db_id
+
+    @property
+    def db_object(self):
+        if self.db_id is None:
+            return None
+        table = self.appliance.db['miq_ae_namespaces']
+        try:
+            return self.appliance.db.session.query(table).filter(table.id == self.db_id)[0]
+        except IndexError:
+            return None
+
+    @property
```

Comment:
Maybe use attrib to make the enabled and locked smart and pulling the correct values from the database when not specified.